cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/repo_integration_test.go

```diff
@@ -711,6 +714,42 @@ func TestKindClusterUpdatePackageRepository(t *testing.T) {
 			expectedDetail:     update_repo_detail_12,
 			failed:             true,
 		},
+		{
+			name:               "update package repository errors when unauthorized",
+			request:            update_repo_req_13,
+			repoName:           "my-podinfo-3",
+			repoUrl:            podinfo_repo_url,
+			expectedStatusCode: codes.PermissionDenied,
+			unauthorized:       true,
+		},
+		{
+			name:               "update url and auth for podinfo package repository (user-managed secrets) errors when secret doesnt exist",
+			request:            update_repo_req_14,
+			repoName:           "my-podinfo-4",
+			repoUrl:            podinfo_repo_url,
+			expectedStatusCode: codes.NotFound,
+			userManagedSecrets: true,
+		},
+		{
+			name:               "update url and auth for podinfo package repository (user-managed secrets)",
+			request:            update_repo_req_14,
+			repoName:           "my-podinfo-4",
+			repoUrl:            podinfo_repo_url,
+			newSecret:          newBasicAuthSecret("secret-1", "TBD", "foo", "bar"),
+			expectedResponse:   update_repo_resp_4,
+			expectedDetail:     update_repo_detail_13,
+			userManagedSecrets: true,
+		},
+		{
+			name:               "update repository change from TLS cert/key to basic auth (kubeapps-managed secrets)",
+			request:            update_repo_req_15,
+			repoName:           "my-podinfo-5",
+			repoUrl:            podinfo_tls_repo_url,
+			oldSecret:          newTlsSecret("secret-1", "TBD", pub, priv, ca),
+			expectedStatusCode: codes.OK,
+			expectedResponse:   update_repo_resp_5,
+			expectedDetail:     update_repo_detail_14,
+		},
```

Comment:
Out of interest: how is the test knowing which use case is user-managed secrets and which is kubeapps-managed secrets? I can't see anything in the use-case here that differentiates (other than the name).

Reply:
Wondering if there's a test showing that the existing secret won't be updated if it's not changed (ie. that the request does not need to contain the actual secret data when not updating the secret - though see my questions in the design doc, I'd understood we were always avoiding sending the secret data back to the UI or requiring it to be sent on update etc., as per your diagram in the appendix, but may have misunderstood that you'd not intended to apply that when Kubeapps is managing secrets, not sure.).

Reply:
> Out of interest: how is the test knowing which use case is user-managed secrets and which is kubeapps-managed secrets? I can't see anything in the use-case here that differentiates (other than the name).

There is a boolean userManagedSecrets in test struct definition

Reply:
> Wondering if there's a test showing that the existing secret won't be updated if it's not changed (ie. that the request does not need to contain the actual secret data when not updating the secret - though see my questions in the design doc, I'd understood we were always avoiding sending the secret data back to the UI or requiring it to be sent on update etc., as per your diagram in the appendix, but may have misunderstood that you'd not intended to apply that when Kubeapps is managing secrets, not sure.).

remember, this is a whole update, not a patch update semantics. No secret data in the update request (regardless of your intention not to update any secret data) => remove any secret data, if present, when handing the update on the backend.

Sounds like what you want is Patch() operation